elastic/kibana
# [Fleet] make removing experimental features more robust

## Summary

Follow up on https://github.com/elastic/kibana/pull/168844
The previous PR tried to make removing experimental features more robust by logging when a feature is removed, unfortunately the logger is not available during config validation.

That PR fix that by adding a warning deprecations when an invalid experimental feature is provided

## How to test

Add this to your config and check a deprecations is logged, and Kibana do not crash

```
xpack.fleet.enableExperimental: ['idonotexists']
```

<img width="1194" alt="Screenshot 2023-10-19 at 4 27 07 PM" src="https://github.com/elastic/kibana/assets/1336873/c06c69d0-132a-474e-872d-c47645b79ff0">
